src/main/java/application/SessionController.java

```diff
@@ -84,74 +95,87 @@ public AnswerQuestionResponseBean answerQuestion(@RequestBody AnswerQuestionRequ
         session.setSequenceId(formEntrySession.getSequenceId() + 1);
         sessionRepo.save(session);
         AnswerQuestionResponseBean responseBean = mapper.readValue(resp.toString(), AnswerQuestionResponseBean.class);
+        responseBean.setSequenceId(formEntrySession.getSequenceId() + 1);
+        log.info("Answer response: " + responseBean);
         return responseBean;
-
     }
 
     @RequestMapping(value = Constants.URL_CURRENT, method = RequestMethod.GET)
     @ResponseBody
     public CurrentResponseBean getCurrent(@RequestBody CurrentRequestBean currentRequestBean) throws Exception {
-        SerializableSession serializableSession = sessionRepo.find(currentRequestBean.getSessionId());
-        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
-        return new CurrentResponseBean(formEntrySession);
+        log.info("Current request: " + currentRequestBean);
+        SerializableFormSession serializableFormSession = sessionRepo.find(currentRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableFormSession);
+        CurrentResponseBean currentResponseBean = new CurrentResponseBean(formEntrySession);
+        log.info("Current response: " + currentResponseBean);
+        return currentResponseBean;
     }
 
     @RequestMapping(value = Constants.URL_SUBMIT_FORM, method = RequestMethod.POST)
     @ResponseBody
     public SubmitResponseBean submitForm(@RequestBody SubmitRequestBean submitRequestBean) throws Exception {
-        SerializableSession serializableSession = sessionRepo.find(submitRequestBean.getSessionId());
-        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
+        log.info("Submit form with bean: " + submitRequestBean);
+        SerializableFormSession serializableFormSession = sessionRepo.find(submitRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableFormSession);
         FormRecordProcessorHelper.processXML(formEntrySession.getSandbox(), formEntrySession.submitGetXml());
-        return new SubmitResponseBean(formEntrySession);
+        SubmitResponseBean submitResponseBean = new SubmitResponseBean(formEntrySession);
+        log.info("Submit response bean: " + submitResponseBean);
```

Comment:
Interesting stuff. Looked into this today and added a filter to get some basic request/response information. Unfortunately it seems like reading the request and response body (what we're really interested in) is more difficult since these are only meant to be read once. So people suggest building a wrapper around this that basically lets you read the input stream and make a copy. This seems like a lot of overhead for logging - especially since we're already doing this with our POJO's which we have access to at the Controller level (which is what a lot of people recommend doing). Not sure if I'm missing some other method here?